File: sorting_algorithms/quicksort/recipe-03/python/testquicksort.py
#!/usr/bin/env python3

"""
File: testquicksort.py

Tests the quicksort algorithm
"""

from quicksort import quicksort
import random

def main(size = 1000, sort = quicksort):
    lyst = []
    for count in range(size):
        lyst.append(random.randint(1, size + 1))
    print(lyst)
    sort(lyst)
    print(lyst)

if __name__ == "__main__":
    main() 

File: sorting_algorithms/quicksort/recipe-02/cxx/testquicksort.cpp
#include <vector>
#include "quicksort.hpp"
#include "random.hpp"
#include "sort_helper.hpp"

int main() {
    int size = 1000;
    std::vector<int> lyst;
    Random random;
    for (int count = 0; count < size; count++) {
        lyst.push_back(random.randint(1, size + 1));
    }
    print(lyst);
    quicksort(lyst);
    print(lyst);
}
